File: BDWebAPI/Models/Entities/Batch.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.InteropServices;

namespace BDWebAPI.Models.Entities
{
    public class Batch
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        public int GroupId { get; set; }

        public int BatchId { get; set; }

        public int TotalProcessedItem { get; set; }

        public int TotalRemainingItem { get; set; }

        public int Total { get; set; }
    }
}

File: BDWebApp/ClientApp/src/app/batch/batch.ts
export interface IBatch {
  batchId?: number;
  totalProcessedItem?: number;
  totalRemainingItem?: number;
  total?: number;
  

}


export interface IBatchOutput extends IBatch{
  currentGroupId?: number,
  isProcessCompleted?: boolean,
  batchList: IBatch[]
}
